WordPress/src/main/java/org/wordpress/android/networking/GlideMShotsLoader.kt

```diff
@@ -32,6 +32,7 @@ class MShot(val url: String)
 class GlideMShotsLoader(noRedirectsRequestQueue: RequestQueue) : ModelLoader<MShot, InputStream> {
     private val requestFactory = VolleyRequestFactory { url, callback, priority, headers ->
         VolleyStreamFetcher.GlideRequest(url, callback, priority, headers).apply {
+            setShouldCache(false) // <- Maybe we should add this here to avoid duplicating cache in Volley?
```

Comment:
I set this false since clearing the Glide cache didn't appear to be enough (I still had to clear the app cache for each test run to see the placeholders and get a cache miss). I suspect that we have _yet another_ cache on the FluxC side, but I haven't tracked it down to confirm that. Meanwhile, I don't have an immediate way to confirm this disables the volley cache via manual testing. The intention here is to disable volley caching just for this usage, rather than at an "app-level", so perhaps updating this comment is all that is needed :thinking: wdyt?

Reply:
I would agree. If I understand this correctly disabling the cache is needed for testing purposes only. Correct?

Reply:
> disabling the cache is needed for testing purposes only. Correct?

Actually, I was considering disabling it on this line in the final changeset. From my understanding, we currently have multiple levels of redundant caching mechanisms:

```mermaid
flowchart LR
  Request --> Glide -.-> Volley -.-> FluxC -.-> Network((Network))
  subgraph Glide
    direction TB
    G1[Request] -.-> Memory -.-> G2[Disk]
  end
  subgraph Volley
    direction TB
    V1[Request] -.-> V2[Disk]
  end
  subgraph FluxC
    direction TB
    F1[Request] -.-> F2[Disk]
  end
```
 What I'm proposing here (with the aim to be effective for this unique usage) is to eliminate the Volley cache layer for this flow:
 
 
```mermaid
flowchart LR
  Request --> Glide -.-> FluxC -.-> Network((Network))
  subgraph Glide
    direction TB
    G1[Request] -.-> Memory -.-> G2[Disk]
  end
  subgraph FluxC
    direction TB
    F1[Request] -.-> F2[Disk]
  end
```

Ideally, we would eliminate the redundant `FluxC` cache layer as well, since `Glide`'s caching is well-implemented, imo. However, that requires some investigation, and possibly additional complexity, since it might not be engineered in a flexible way to readily support this. Regarding `Volley`, since we already "incur" the complexity (wired up to avoid the `MShots` redirect placeholder image being cached in `Glide`), disabling the extra cache layer can be accomplished with with 1 line, afaik. But, I was unable to actually test this properly, because of the additional cache layers :disappointed: .

Though `setShouldCache` is a `Volley` method, I was uncertain whether this flag somehow affects `Glide` through `Glide`/`Volley` integration. I was a bit surprised to learn that after including this line, I _still_ needed to clear the app cache to get the fresh experience. I'm not 100% certain that the additional caching is in `FluxC`, but it is a hunch.

Reply:
Since testing this may not be feasible at the moment, maybe the best course is to remove this line and add this as a task for disk-usage optimization, to be tackled at a later point. Wdyt?

Reply:
I removed this line here: https://github.com/wordpress-mobile/WordPress-Android/pull/16584/commits/0cfce3d95cf1c00daa4009d2d25a77bbf0cf2967 and we can re-add / revisit if needed.